lib/Extractor/Solver.cpp

```diff
@@ -206,8 +206,24 @@ class BaseSolver : public Solver {
     }
 
     if (InferInsts && SMTSolver->supportsModels()) {
+      std::vector<Inst *> LHSComps;
+      findCands(LHS, LHSComps, IC, MaxNops);
+
+      std::vector<Inst *> Copies;
+      std::vector<InstMapping> PCsCopy;
+      BlockPCs BPCsCopy;
+      for (auto I : LHSComps) {
+        // separate sub-expressions by copying vars
+        std::map<Inst *, Inst *> InstCache;
+        std::map<Block *, Block *> BlockCache;
+        Copies.push_back(getInstCopy(I, IC, InstCache, BlockCache));
+        separateBlockPCs(BPCs, BPCsCopy, InstCache, BlockCache, IC);
+        separatePCs(PCs, PCsCopy, InstCache, BlockCache, IC);
+      }
+
       InstSynthesis IS;
-      EC = IS.synthesize(SMTSolver.get(), BPCs, PCs, LHS, RHS, IC, Timeout);
+      EC = IS.synthesize(SMTSolver.get(), BPCs, PCs, LHS, RHS,
+                         Copies, PCsCopy, BPCsCopy, IC, Timeout);
```

Comment:
There is no need to make copies, the only thing I need are the ``LHSComps``.

Reply:
Wait, really? But we need copies for nop synthesis, what's different here?